modules/otel-collector-deployment-modes.adoc

```diff
@@ -0,0 +1,49 @@
+//Module included in the following assemblies:
+//
+// * observability/otel/otel-collector/otel-collector-configuration-intro.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="otel-collector-deployment-modes_{context}"]
+= Deployment modes
+
+The OpenTelemetry Collector supports the following deployment modes:
+
+* Using the relevant Kubernetes workload resource:
++
+Deployment:: The default.
++
+StatefulSet:: If you need to run stateful workloads, for example when using the Collector's File Storage Extension or Tail Sampling Processor, you can choose the StatefulSet deployment mode.
++
+DaemonSet:: If you need to scrape telemetry data from every node, for example by using the Collector's Filelog Receiver to read container logs, you can choose the DaemonSet deployment mode.
```

Comment:
I suggest to add a link for the filelog recevier

Reply:
Unfortunately, we are not allowed to put any text into assembly files, and we are also not allowed to put links inside module files, and I am required to move all the content that has been in assembly files until now into module files as preparation for migrating the files to DITA. On top of that all, the management's ask is to minimize the use of links wherever possible.